### PR TITLE
Updates guide configs

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?

Reformats guide configs to allow reading from a `generated` source.

## Notes to reviewers

I don't understand the links file well enough to go through and do this.

```
Exhaustive list of all broken links found:

- On source page path = /activities:
   -> linking to /go/how-to-develop-an-activity-definition-in-go

- On source page path = /dev-guide/go:
   -> linking to /go/backgroundcheck-boilerplate-run-a-dev-server-worker
   -> linking to /go/backgroundcheck-boilerplate-cloud-worker
   -> linking to /go/backgroundcheck-boilerplate-add-test-framework

- On source page path = /dev-guide/go/foundations:
   -> linking to /go/how-to-develop-an-activity-definition-in-go
   -> linking to /go/how-to-develop-a-worker-in-go
   -> linking to /go/how-to-customize-workflow-type-in-go
   -> linking to /go/how-to-customize-activity-type-in-go

- On source page path = /dev-guide/python:
   -> linking to /python/backgroundcheck-boilerplate-add-test-framework

- On source page path = /workers:
   -> linking to /go/how-to-develop-a-worker-in-go

- On source page path = /workflows:
   -> linking to /python/how-to-use-the-python-sdk-patching-api-in-python
 ```

### CODE REVIEW

File: assembly/activities/attachSourceToGuides.js [hunk 1]
```diff
@@ -59,11 +59,37 @@ async function matchLangTabs(langtabs, infoNodes) {
...
+      });
+
+      // Diagnostic: Print info when searching within 'generated' folder
+      if (match) {
+        console.log(`Match found in 'generated' folder for id: ${generatedId}`);
+      } else {
+        console.log(`No match found in 'generated' folder for id: ${generatedId}`);
+      }
+    }
+  }
```
Comment (flossypurse) [inline assembly/activities/attachSourceToGuides.js:87]: Has this been helpful for you in finding where the right node id lives?

File: assembly/activities/genSourceObjects.js [hunk 2]
```diff
@@ -8,14 +8,21 @@ export async function genSourceObjects(config) {
...
       }
     }
+
+    // Check inside the 'generated' folder
+    const generatedDir = path.join(readDir, 'generated');
+    for await (const generatedEntry of readdirp(generatedDir)) {
+      if (path.extname(generatedEntry.fullPath) == ".md") {
+        filePaths.push(generatedEntry);
+      }
+    }
```
Comment (flossypurse) [inline assembly/activities/genSourceObjects.js:25]: The code above already gets all nested nodes / files.

For example, check out the `docs-src/cloud/tcld`, or `docs-src/cli` directories.

File: assembly/activities/linkMagic.js [hunk 3]
```diff
@@ -123,14 +123,22 @@ async function replaceLinks(line, match, link, current_guide_id) {
...
     localRef = match[3];
   }
-  // define the new path
 
+  // Check if the link is inside the 'generated' folder
+  const segments = link.node_id.split('/');
+  if (segments.length > 2 && segments[1] === 'generated') {
+    link.file_dir = `${segments[0]}/generated`;
+    link.guide_id = segments.slice(2).join('/');
+  }
```
Comment (flossypurse) [inline assembly/activities/linkMagic.js:139]: This shouldn't be required.

LinkMagic already works with nested directories.

File: delete.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+# Iterate over all .md files in the root directory
+for file_path in root_dir.glob('*.md'):
+    generated_file_path = generated_dir / file_path.name
+
+    # If the file exists in both directories, delete it from the root directory
+    if generated_file_path.exists():
+        file_path.unlink()
+
+print("Finished checking and deleting duplicate files.")
```
Comment (flossypurse) [inline delete.py:16]: I am not sure I want to rely on this for curating the nodes...

Let's chat about the need for this.

Comment (rachfop) [reply delete.py:16]: This was a one-off script I made to delete duplicated files.

**Pull request closed without merging**